libraries/apollo-runtime-java/src/main/java/com/apollographql/apollo3/runtime/java/interceptor/ApolloInterceptor.java

```diff
@@ -1,24 +1,23 @@
 package com.apollographql.apollo3.runtime.java.interceptor;
 
 import com.apollographql.apollo3.api.ApolloRequest;
+import com.apollographql.apollo3.api.Operation;
 import com.apollographql.apollo3.runtime.java.ApolloCallback;
 import org.jetbrains.annotations.NotNull;
 
-import java.util.concurrent.Executor;
-
 /**
- * ApolloInterceptor is responsible for observing and modifying the requests going out and the corresponding responses
- * coming back in. Typical responsibilities include adding or removing headers from the request or response objects,
- * transforming the returned responses from one type to another, etc.
+ * ApolloInterceptor is responsible for observing and modifying the requests going out and the corresponding responses coming back in.
+ * Typical responsibilities include adding or removing headers from the request or response objects, transforming the returned responses
+ * from one type to another, etc.
  */
-public interface ApolloInterceptor {
+public interface ApolloInterceptor<D extends Operation.Data> {
```

Comment:
I'm curious how come that works? A single `ApolloInterceptor` typically sees different `D` so it cannot really be parametrized?

Reply:
Woops you're probably right, I went too fast on this 😅. Currently the interceptors list is created inside `execute`, but later it should be in `ApolloClient` and not specific to one `D`.

Reply:
I think it should be good now in 7177271d98a633e1fdc6f42a0ace3d1f1807ddbb